lib/zclient.h

```diff
@@ -348,6 +350,19 @@ struct zclient {
 	void (*zebra_connected)(struct zclient *);
 	void (*zebra_capabilities)(struct zclient_capabilities *cap);
 
+	/*
+	 * match -> is the prefix that the calling daemon asked to be matched
+	 * against.
+	 * nhr->prefix -> is the actual prefix that was matched against in the
+	 * rib itself.
+	 *
+	 * This distinction is made because a LPM can be made if there is a
+	 * covering route.  This way the upper level protocol can make a
+	 * decision point about whether or not it wants to use the match or not.
+	 */
+	void (*nexthop_update)(struct vrf *vrf, struct prefix *match,
```

Comment:
can 'match' and 'nhr' be 'const'?

Reply:
I tried that and it had a trail of fallout (from other functions taking non-const args) behind it that I didn't want to shove into this PR to keep it clean-ish :disappointed: